src/main/java/Duke.java

```diff
@@ -1,10 +1,117 @@
+import DukePackage.ChatUI;
+import DukePackage.Parser;
+import DukePackage.Storage;
+import DukePackage.Task;
+import DukePackage.TaskType;
+import java.util.Objects;
+
+
+/**
+ * The Duke class is the main class that runs the Duke chatbot application.
+ */
 public class Duke {
+
+    /**
+     * The main method that starts the Duke application.
+     * @param args The command-line arguments passed to the application.
+     */
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        // standard response
+        String horizontalLine = "   ------------------------------------------------------------------------";
+        String noDescError = "     ☹ OOPS!!! The description of a todo cannot be empty.";
+        String noCommandError = "     ☹ OOPS!!! I'm sorry, but I don't know what that means :-(";
+
+        // initialise
+        Storage storage = new Storage();
+        Parser parser = new Parser();
+        ChatUI ui = new ChatUI();
+
+        // read from txt file and create tasks and put into storage
+        storage.loadListFromFile();
+        ui.printIntro();
+
+        while (true) {
+            String input = parser.getInput();
+            String[] parts = input.split(" ");
+            ui.printDivider();
+
+            switch (parts[0]) {
+                case "bye":
+                    ui.printOutro();
+                    ui.printDivider();
+                    // write the changes into the file duke.txt
+                    storage.writeTasksToFile();
+                    return;
+                case "list":
+                    storage.printTaskList();
+                    break;
+                case "mark":
+                    int id = Integer.parseInt(parts[1]) - 1;
+                    try {
+                        storage.changeTaskMarking(id, true);
+                        storage.printTaskMarking(id);
+                    } catch (Exception e) {
+                        System.out.print(e.getMessage());
```

Comment:
Hmm, perhaps you can encapsulate the printing of exceptions thrown under ChatUI class as well?